log/value.go

```diff
@@ -39,6 +39,36 @@ func Timestamp(t func() time.Time) Valuer {
 	return func() interface{} { return t() }
 }
 
+// TimestampFormat returns a Valuer that produces a TimeFormat value from
+// layout and the time returned by t. Users will probably want to use
+// DefaultTimestamp or DefaultTimestampUTC.
+func TimestampFormat(t func() time.Time, layout string) Valuer {
+	return func() interface{} {
+		return TimeFormat{
+			Time:   t(),
+			Layout: layout,
+		}
+	}
+}
+
+// A TimeFormat represents an instant in time and a layout used when
+// marshaling to a text format.
```

Comment:
- [ ] If this does stay an exported type, consider mentioning that the layout must be one accepted by [`time.Time.Format`](https://golang.org/pkg/time/#Time.Format).